### PR TITLE
🌱 add hivelocity infra provider to clusterctl

### PR DESCRIPTION
**Pull request by kranurag7**
**What this PR does / why we need it**:

- adds hivelocity infra provider to clusterctl 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/10167

/area clusterctl

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/ok-to-test

File: cmd/clusterctl/client/config/providers_client.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ const (
 	DOProviderName             = "digitalocean"
 	GCPProviderName            = "gcp"
 	HetznerProviderName        = "hetzner"
+	HivelocityProviderName     = "hivelocity"
```
Comment (chrischdi) [inline cmd/clusterctl/client/config/providers_client.go:48]: To follow https://main.cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#adding-a-provider-to-clusterctl : the provider should contain the github org name so if I'm right in this case it should be `hivelocity-hivelocity` .

Comment (kranurag7) [reply cmd/clusterctl/client/config/providers_client.go]: The test is failing after adding the prefix. 

From docs linked above: 
_The need to add a prefix for providers not in the kubernetes-sigs org applies to all the providers being added to clusterctl‘s pre-defined list of provider starting from January 2024._

After reading the docs, I don't have full clarity on what the prefix should be?

Comment (kranurag7) [reply cmd/clusterctl/client/config/providers_client.go]: @chrischdi After I migrated the PR to use the new naming convention suggested above, the tests are failing, it was not failing when I used only `hivelocity` as provider name. Is there something else I should be aware of? 
Also is the prefix `hivelocity` correct in this case or this should be something else?

Comment (chrischdi) [reply cmd/clusterctl/client/config/providers_client.go]: My guess is: the tests were not touched yet when introducing the new naming schema, this is the first provider which gets added with the new schema I guess.

Comment (kranurag7) [reply cmd/clusterctl/client/config/providers_client.go]: I rebased the PR against main and then updated as per the above PR. The tests are failing with the following message. 

```bash
Gomega truncated this representation as it exceeds 'format.MaxLength'.
        Consider having the object provide a custom 'GomegaStringer' representation
        or adjust the parameters in Gomega's 'format' package.
        
        Learn more here: https://onsi.github.io/gomega/#adjusting-output
```
I tested this on main branch (cd cmd/clusterctl/cmd/; go test -run Test_runGetRepositories) and on main this error is not there.

File: cmd/clusterctl/cmd/config_repositories_test.go [hunk 2]
```diff
@@ -130,6 +130,7 @@ digitalocean           InfrastructureProvider   https://github.com/kubernetes-si
 docker                 InfrastructureProvider   https://github.com/kubernetes-sigs/cluster-api/releases/latest/                             infrastructure-components-development.yaml
 gcp                    InfrastructureProvider   https://github.com/kubernetes-sigs/cluster-api-provider-gcp/releases/latest/                infrastructure-components.yaml
 hetzner                InfrastructureProvider   https://github.com/syself/cluster-api-provider-hetzner/releases/latest/                     infrastructure-components.yaml
```
Comment (chrischdi) [inline cmd/clusterctl/cmd/config_repositories_test.go:132]:
```suggestion
docker                  InfrastructureProvider   https://github.com/kubernetes-sigs/cluster-api/releases/latest/                             infrastructure-components-development.yaml
gcp                     InfrastructureProvider   https://github.com/kubernetes-sigs/cluster-api-provider-gcp/releases/latest/                infrastructure-components.yaml
hetzner                 InfrastructureProvider   https://github.com/syself/cluster-api-provider-hetzner/releases/latest/                     infrastructure-components.yaml
```

**Comment by kranurag7** [comment]
> /tide merge-method-squash

@chrischdi I think in this case manual squashing is not required from my end. I was thinking of doing it, once the checks are green.

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**